crates/teloxide/src/dispatching/tracing.rs

```diff
@@ -0,0 +1,34 @@
+use super::{DpHandlerDescription, UpdateHandler};
+
+use dptree::{di::Injectable, prelude::DependencyMap, HandlerDescription};
+use tracing::{Instrument, Span};
+
+/// Extension methods for adding `tracing` to [`UpdateHandler`].
+///
+/// This trait is only available if the `tracing` feature is enabled.
+pub trait UpdateHandlerTracingExt<E> {
+    /// Returns an `UpdateHandler` with tracing enabled.
+    fn with_tracing_span<F, FnArgs>(self, f: F) -> Self
```

Comment:
Can you rename this to `instrument_with_async` + add `instrument_with` which is the same, but uses `Asyncify<F>: ` bound? (see the `dptree` for how async/sync versions are implemented).